assets/cart.js

```diff
@@ -42,7 +42,8 @@ class CartItems extends HTMLElement {
   }
 
   onChange(event) {
-    this.updateQuantity(event.target.dataset.index, event.target.value, document.activeElement.getAttribute('name'));
+    const newQuantity = event.target.max ? Math.min(event.target.value, event.target.max) : event.target.value;
+    this.updateQuantity(event.target.dataset.index, newQuantity, document.activeElement.getAttribute('name'));
```

Comment:
Hey sure thing @ludoboludo and I apologize my description wasn't specific enough. This actually happens in the cart drawer and on the cart page quantity inputs. If you add the max to the cart on the product page, and then manually change the quantity input to a number greater than the max and blur the input by clicking away, then I'm able to actually adjust the cart quantity to a number greater than the max (on the cart page and in the cart drawer). I'll update the PR accordingly with some clearer steps to reproduce this.

Reply:
Even with those replication steps I can't make it happen on my end: [video](https://screenshot.click/24-22-40u8s-5q54b.mp4)
There were some changes recently around quantity rules and what not. Could you confirm you're testing on the latest version of the repo ? Or let me know if there is something I'm missing in my product setup

Reply:
Hey @ludoboludo , I haven't tried with the latest bleeding edge pull yet, but I think this may be happening on my end because I'm actually setting the quantity max *based on location*. Long story short, I have a Flow that runs and updates a metafield for each variant so the metafield contains a location based inventory, then I set the quantity input max based on which location the visitor is actually shopping for. So, often times the input max is actually less than the *total* inventory quantity. When that happens, there isn't any API validation of the quantity. To replicate that in your example where the product has a total inventory quantity of 2 available, try manually setting the input max to just 1. When you do that, I believe you'll be able to then change the input in the drawer and on the cart page to 2, and thus you'll be able to circumvent the form's `max` attribute (because the API validation won't kick in until qty > 2). Lmk if that helps!

Reply:
Ah I see. But then that kinda make sense to me as it's basing it's validation on the API and the inventory quantity its seeing for that product. Not based on the HTML attribute assign to the input 🤔 

If we wanted to change the approach so it's based on the input attribute we would need to change the `updateQuantity(line, quantity, name){...}` function to look at that attribute value instead of through API/fetch validation. 
I don't think there is a way to check specifically for a location which could be potentially useful. 

So in your theme are you having different line items for the same product if they're coming from different inventory locations ?